db/db_impl.cc

```diff
@@ -1225,7 +1225,7 @@ Status DBImpl::GetImpl(const ReadOptions& read_options,
     // TODO: This mutex should be removed later, to improve performance when
     // tracing is enabled.
     InstrumentedMutexLock lock(&trace_mutex_);
-    if (tracer_) {
+    if (tracer_ && !tracer_->IsTraceFileOverMax()) {
```

Comment:
Can you move this logic to tracer_->Get()? I hope we minimize the instruction we added to this function.

Reply:
Sure, others are also moved to tracer_